test/nexus_test.go

```diff
@@ -427,16 +422,19 @@ func TestSyncOperationFromWorkflow(t *testing.T) {
 		switch outcome {
 		case "successful":
 			return outcome, nil
-		case "failed":
-			return "", &nexus.UnsuccessfulOperationError{
-				State:   nexus.OperationStateFailed,
-				Failure: nexus.Failure{Message: "failed for test"},
+		case "failed-plain-error":
+			return "", nexus.NewFailedOperationError(errors.New("failed for test"))
+		case "failed-app-error":
```

Comment:
What are these cases for? They don't appear to be called by any test

Reply:
I left these and didn't commit the tests since we don't have server support yet. The tests are in a branch on my machine. I can start a draft PR that we'll merge once CLI with 1.26.2 is out.